File: dsa_problems/convertToindianCurr.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    double n;
    cout<<"Enter the amount in dollars: ";
    cin>>n;

    string str = to_string(n);
    bool isDecimal = false;
    string lastPart = "";
    string firstPart = "";


    for(int i=0;i<str.size();i++){
        if(str[i] == '.' || isDecimal){
            lastPart += str[i];
            isDecimal = true;
        }
        else{
            firstPart += str[i];
        }
    }

    if(firstPart.size() > 3){
        reverse(firstPart.begin(), firstPart.end());
        string temp = "";
        for(int i=0;i<firstPart.size();i++){
            temp += firstPart[i];
            if(i%2 == 0 && i != 0 && i != firstPart.size()-1){
                temp += ',';
            }
        }
        reverse(temp.begin(), temp.end());
        firstPart = temp;
    }

    firstPart += lastPart;
    cout<<"Amount in Indian currency: Rs. "<<firstPart<<endl;
    return 0;
}

File: dsa_problems/convertToindianCurr.py
# Convert Doller to Indian Currency

n = float(input("Enter the amount in dollars: "))
    
str_n = str(n)
is_decimal = False
last_part = ""
first_part = ""

for char in str_n:
    if char == '.' or is_decimal:
        last_part += char
        is_decimal = True
    else:
        first_part += char

if len(first_part) > 3:
    first_part = first_part[::-1]
    temp = ""
    for i in range(len(first_part)):
        temp += first_part[i]
        if i % 2 == 0 and i != 0 and i != len(first_part) - 1:
            temp += ','
    first_part = temp[::-1]

first_part += last_part
print(f"Amount in Indian currency: Rs. {first_part}")